### PR TITLE
tasks/DES-2655: Select file modal in Apps Form

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview: ##
Add select file modal. Most of the modal design is a trimmed out and adjusted version of CopyModal from DataFiles.

## PR Status: ##

* [x] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2655](https://tacc-main.atlassian.net/browse/DES-2655)

## Summary of Changes: ##
* Use modal similar to CopyModal - Just one panel of file listing.
* Use Systems list to derive the list of available systems.
* Always use tapis as api
* Integrate into FormItem.

## Testing Steps: ##
1. Go to app: https://designsafe.dev/rw/workspace/openfoam#!/
2. Select file and submit the job.

## UI Photos:
Screenshot 2024-05-22 at 4 06 50 PM

community data:

Screenshot 2024-05-23 at 11 23 06 AM

system dropdown:
Screenshot 2024-05-23 at 11 23 11 AM

## Notes: ##
* The system listing needs some work. Dynamically generating the list has some gaps - missing labels, duplication.

### CODE REVIEW

File: client/modules/workspace/src/SelectModal/SelectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    label: system.notes?.label ?? system.id,
+    value: system.id,
+  }));
+  systemOptions.push({ label: 'My Projects', value: 'myprojects' });
+
+  const defaultParams = useMemo(
+    () => ({
+      selectedApi: api,
+      selectedSystem: defaultStorageSystem.id,
+      selectedPath: encodeURIComponent('/' + user?.username),
```
Comment (rstijerina) [inline client/modules/workspace/src/SelectModal/SelectModal.tsx:108]: Going to Community Data navigates to `Community Data / sal`, which I think is unintended

Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:108]: oh, will fix.

File: client/modules/workspace/src/SelectModal/SelectModal.tsx [hunk 2]
```diff
@@ -35,6 +37,20 @@ const HeaderTitle: React.FC<{
...
+const getScheme = (storageSystem: TTapisSystem): string => {
+  return storageSystem.isDynamicEffectiveUser ? 'private' : 'public';
+};
+
+const getPath = (
+  storageSystem: TTapisSystem,
+  user: TUser | undefined
+): string => {
+  return storageSystem.isDynamicEffectiveUser
+    ? encodeURIComponent('/' + user?.username)
```
Comment (rstijerina) [inline client/modules/workspace/src/SelectModal/SelectModal.tsx:50]: I wonder if there could be cases in the future where `isDynamicEffectiveUser` is not a good enough flag to assume a child dir with the user's username is present. Perhaps we should add another field to `notes` that's `isMyData` to denote that a username child dir is expected

Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:50]: Adjusted the logic

Comment (chandra-tacc) [reply client/modules/workspace/src/SelectModal/SelectModal.tsx:50]: Re-tested it.

**Review by rstijerina — APPROVED** [review]
🚀

**Pull request merged**